eland/ml/exporters/es_gb_models.py

```diff
@@ -0,0 +1,449 @@
+
+        for field_name in field_names:
+            if field_name in feature_names and field_name not in input_field_names:
+                input_field_names.add(field_name)
+
+        return feature_names, input_field_names
+
+    def fit(self, X, y, sample_weight=None, monitor=None) -> None:
+        """
+        Override of the sklearn fit() method. It does nothing since Elastic ML models are
+        trained in the Elastic Stack or imported.
+        """
+        # Do nothing, model if fitted using Elasticsearch API
+        pass
+
+    @property
+    @abstractmethod
+    def analysis_type(self):
+        """
+        Type of the data frame analysis. It can be classification or regression.
+        """
+        pass
+
+    @abstractmethod
+    def _initialize_init_(self) -> Union[DummyClassifier, DummyRegressor]:
+        pass
+
+
+class ESGradientBoostingClassifier(ESGradientBoostingModel, GradientBoostingClassifier):
+    """
+    Elastic ML model wrapper compatible with sklearn GradientBoostingClassifier.
+    """
+
+    def __init__(
+        self,
+        es_client: Union[str, List[str], Tuple[str, ...], "Elasticsearch"],
+        model_id: str,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        es_client : Elasticsearch client argument(s)
+            - elasticsearch-py parameters or
+            - elasticsearch-py instance
+        model_id : str
+            The unique identifier of the trained inference model in Elasticsearch.
+
+        Raises
+        ------
+        NotImplementedError
+            Multi-class classification is not supported at the moment.
+        ValueError
+            The classifier should be defined for at least 2 classes.
+        """
+        ESGradientBoostingModel.__init__(self, es_client, model_id)
+        self._extract_common_parameters()
+        GradientBoostingClassifier.__init__(
+            self,
+            learning_rate=1.0,
+            n_estimators=self.n_estimators,
+            max_depth=self._max_depth,
+        )
+
+        if "classification_labels" in self._definition["trained_model"]["ensemble"]:
+            self.classes_ = np.array(
+                self._definition["trained_model"]["ensemble"]["classification_labels"]
+            )
+        else:
+            self.classes_ = None
+
+        self.n_outputs = self._n_outputs
+        if self.classes_ is not None:
+            self.n_classes_ = len(self.classes_)
+        elif self.n_outputs <= 2:
+            self.n_classes_ = 2
+        else:
+            self.n_classes_ = self.n_outputs
+
+        if self.n_classes_ == 2:
+            self._loss = BinomialDeviance(self.n_classes_)
+            # self.n_outputs = 1
+        elif self.n_classes_ > 2:
+            raise NotImplementedError("Only binary classification is implemented.")
+        else:
+            raise ValueError(f"At least 2 classes required. got {self.n_classes_}.")
+
+        self.init_ = self._initialize_init_()
```

Comment:
Is `self.init_` used anywhere? It looks like we're returning an estimator or something like this, if it is needed we should use a better name.

Reply:
`self.init_`  is a part of `sklearn` estimator definition. It is usually initialized when `fit()` is called. In our case, we don't use `fit()` since models are trained within Elasticsearch. Hence, I need to initialize it manually. `sklearn` is then uses `self.init_` for predictions, etc.